tests/miral/config_file.cpp

```diff
@@ -89,13 +92,23 @@ struct TestConfigFile : PendingLoad, miral::TestServer
 
     void SetUp() override
     {
+        start_server_in_setup = false;
         miral::TestServer::SetUp();
         ON_CALL(*this, load(testing::_, testing::_)).WillByDefault([this]{ notify_load(); });
+
+        invoke_runner(
+            [&](auto& runner)
+            {
+                runner.add_stop_callback(
+                    [&]
+                    {
+                        reloading_config_file.reset();
+                    });
+            });
```

Comment:
There's clearly a synchronization issue, but simply changing the timing this way and "hoping for the best" isn't a fix.